### PR TITLE
Fix progress aborted by callback

### PR DESCRIPTION
**Pull request by consatan**
`CURLOPT_PROGRESSFUNCTION` return a non-zero value to abort the transfer. In which case, the transfer will set a `CURLE_ABORTED_BY_CALLBACK` error. 

`Request` option `progress` cannot abort the transfer now.

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Thanks for your contribution, I will review it shortly.

**Comment by consatan** [comment]
ping?

**Comment by MurgaNikolay** [comment]
Hello. Any updates?

**Comment by iwo-borczuch** [comment]
any possibility to merge in near future?

**Comment by BenMorel** [comment]
@sagikazarmark When is *soon*?

**Comment by Nyholm** [comment]
Thank you for this PR. Could you rebase on master please?

**Comment by consatan** [comment]
@Nyholm 

Completed, the BC Check failing, but look like is not my fault.

File: src/Handler/CurlFactory.php [hunk 1]
```diff
@@ -484,7 +487,7 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf)
                 if (is_resource($args[0])) {
                     array_shift($args);
                 }
-                call_user_func_array($progress, $args);
+                return call_user_func_array($progress, $args);
```
Comment (Nyholm) [inline src/Handler/CurlFactory.php:490]: Why is this needed?

Comment (consatan) [reply src/Handler/CurlFactory.php:490]: The `CURLOPT_PROGRESSFUNCTION` option return a non-zero value to abort the transfer.

https://www.php.net/manual/en/function.curl-setopt.php

> Return a non-zero value to abort the transfer. In which case, the transfer will set a CURLE_ABORTED_BY_CALLBACK error.

**Comment by GrahamCampbell** [comment]
@consatan Could you rebase again, please?
